packages/grafana-schema/src/schema/mudball.cue

```diff
@@ -265,3 +265,34 @@ VizTooltipOptions: {
 	mode: TooltipDisplayMode
 	sort: SortOrder
 } @cuetsy(kind="interface")
+
+
+// These are the common properties available to all queries in all datasources.
+// Specific implementations will *extend* this interface, adding the required
+// properties for the given context.
+DataQuery: {
```

Comment:
**Question**: Do we have a plan how to handle changes over time for these kind of types?

Reply:
yes, these have to be strictly controlled - only forwards _and_ backwards compatible changes are permissible.

Reply:
A checker to ensure that invariant is prerequisite to being able to mark any of our individual kinds as stable.